test_threadleak.py

```diff
@@ -64,7 +73,7 @@ def test_leak_enabled_option(testdir):
         '*::test_leak FAILED*',
         '*Failed: Test leaked *leaked-thread-0*leaked-thread-1*',
     ])
-    assert result.ret == 1
+    assert result.ret == ExitCode.TESTS_FAILED
```

Comment:
I don't see much value in this change. It can make sense if the enums are public. Otherwise using 0 and 1 is fine. If you really want to use enums and there is no way to import them from pytest without accessing _private packages, you can define them in this file.

This change is also not related to the new feature, so it is best done in another PR.